### PR TITLE
React & Redux capstone project

### PR DESCRIPTION
**Pull request by EddyShimwa**
### I,
- Used React documentation.
- Used React components.
- Used React props.
- Used React Router.
- Connected React and Redux.
- Wrote integration tests with a React testing library.
- Used styles in a React app.
- Used React life cycle methods.
- Applied React best practices and language style guides in code.
- Used store, actions and reducers in React.
- Recorded a Room video
- I wrote the testes and they are all passing
- No linter errors
Here are some of the Images of my app
image1

image2

- Testes
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,16 +51,24 @@
...
 
-## 🚀 Live Demo <a name="live-demo"></a>
+> ## 🚀 Live Demo <a name="live-demo"> </a>
+- [Check out the live demo here](https://explore-america.onrender.com)
 
 > ## Project demo
->
-> Site is under construction and the Project Demo will update soon.
+[Check out the live video here](https://www.loom.com/share/678ae641b5fe4d64b231a424bcfefbbc)
+
```
Comment (jerryowusu) [inline README.md]: Well done for making the video presentation, however, as part of the project requirements, you should also highlight some interesting piece of code or something you built that you are very proud of  in the video. Kindly check and update that. :wink:

File: src/Redux/Country/countrySlice.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-param-reassign */
```
Comment (jerryowusu) [inline src/Redux/Country/countrySlice.js:1]: Linters are there to help you achieve standard best practices, so it is not the best to disable any of them. Kindly make sure you remove this line. :+1:

File: src/Components/homepage.css [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ul {
+  padding: 0%;
+}
+
+.country-name {
+  width: 100%;
+  display: flex;
+  list-style: none;
+  align-items: center;
+}
```
Comment (jerryowusu) [inline src/Components/homepage.css]: Nice and beautiful design so far, however, as stated in the project requirements, you are suppose to follow the design guidelines including: 

- [x] Colors (select one main color for your website).
- [x] Typography: font face, size and weight.
- [x] Layout: composition and space between elements.

There are some differences between yours and that of the actual design which you will need to fix.

1.  The home page is actually suppose to be in two rows with a light and dark color alternating between the cards on each column.  
2. Also, there are no spaces between the cards
3. Kindly check the position and size of the country image/flag, country name, as well as the small arrows on the top right corner of each country card.



### Your design 


Screenshot from 2023-02-09 22-48-49


### The actual design 

Screenshot from 2023-02-07 18-49-55

File: src/Components/DetailsPage.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+              {' '}
+              {country.population}
+            </li>
+            <button type="submit" className="view"><a href="{country.maps.googleMaps}">View on map</a></button>
+          </ul>
+        </div>
+      ) : (
+        <div className="loading">Loading...</div>
+      )}
+    </div>
```
Comment (jerryowusu) [inline src/Components/DetailsPage.js]: Nice and beuatiful design so far, however, as stated in the project requirements, you are suppose to follow the design guidelines including: 

- [x] Colors (select one main color for your website).
- [x] Typography: font face, size and weight.
- [x] Layout: composition and space between elements.

There are some differences between yours and that of the actual design which you will need to fix.

1. The background of each odd card is darker than its preceeding (even) card forming a checkered layout.
2. Also, kindly note the margin, paddings, and distance between the cities name (your green text) and the views (your white text)
3. Your design is also missing the small arrow to the right of each card. 

### Your design 

Screenshot from 2023-02-09 22-58-58


### The actual design 

Screenshot from 2023-02-07 18-50-02

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @EddyShimwa , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost-paul-walker


## Highlights
Great effort so far :+1: 
No linter errors :+1: 

## Required Changes ♻️

- [ ] The link to the license is broken since your app does not contain any "LICENSE" file path that you have referenced. Kindly fix this issue

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EddyShimwa** [comment]
Thanks @jerryowusu for great suggestions to ma make my project look more better! I appreciate!
I am going to make all the changes changes you suggested

**Comment by EddyShimwa** [comment]
Here, I made all the changes suggested by the reviewer

File: src/Components/homepage.css (hunk 3 above)
Comment (zieeco) [reply src/Components/homepage.css]: - [ ] This issue is well resolved :heavy_check_mark:

File: src/Components/DetailsPage.js (hunk 4 above)
Comment (zieeco) [reply src/Components/DetailsPage.js]: - [ ] This issue is well resolved :heavy_check_mark:

File: README.md (hunk 1 above)
Comment (zieeco) [reply README.md]: - [ ] This issue is well resolved :heavy_check_mark:

File: README.md [hunk 5]
```diff
@@ -151,7 +159,7 @@ Feel free to check the [issues page](../../issues/).
 
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
-> Without Microverse Help this project not compleated at all so thanks a lot Microverse for giving us this kind of opertunity.
+> Specials thanks Nelson Sakwa who came up with this amazing original design
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (zieeco) [inline README.md:164]: ### README LICENCE 

- [x] This issue is not yet resolved :heavy_multiplication_x: 

   - As stated by the previous reviewer, please, kindly add a licence to your readme to make it more professional as the link is currently broken.

:point_down: 
*previous reviewer's required changes*
Screenshot from 2023-02-10 15-27-42

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @EddyShimwa ,


Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 5 above)
Comment (EddyShimwa) [reply README.md:164]: Thanks a lot @zieeco

**Review by pinkmoon25 — APPROVED** [review]
Hi @EddyShimwa , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linters :heavy_check_mark: 
GitHub flow :heavy_check_mark: 
Good design :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**